src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -1397,24 +1401,38 @@ void CountryInstance::tick(InstanceManager& instance_manager) {
 	//income_from_gold = country_defines.gold_to_cash_rate * sum (total_good_production * good.get_base_price()) for each good with is_money=true
 }
 
+void CountryInstance::good_data_t::clear_daily_recorded_data() {
+	const std::lock_guard<std::mutex> lock_guard {*lock};
+	stockpile_change_yesterday
+		= exported_amount
+		= government_needs
+		= army_needs
+		= navy_needs	
+		= production_needs
+		= overseas_needs
+		= factory_needs
+		= pop_needs
+		= available_amount
+		= fixed_point_t::_0();
+	need_consumption_per_pop_type.clear();
+	input_consumption_per_production_type.clear();
+	production_per_production_type.clear();
+}
 
 void CountryInstance::report_pop_need_consumption(PopType const& pop_type, GoodDefinition const& good, const fixed_point_t quantity) {
-	//TODO
-}
-void CountryInstance::report_artisan_input_consumption(ProductionType const& production_type, GoodDefinition const& good, const fixed_point_t quantity) {
-	//TODO
-}
-void CountryInstance::report_artisan_output(ProductionType const& production_type, const fixed_point_t quantity) {
-	//TODO
-}
-void CountryInstance::report_factory_input_consumption(ProductionType const& production_type, GoodDefinition const& good, const fixed_point_t quantity) {
-	//TODO
-}
-void CountryInstance::report_factory_output(ProductionType const& production_type, const fixed_point_t quantity) {
-	//TODO
-}
-void CountryInstance::report_rgo_output(GoodDefinition const& good, const fixed_point_t quantity) {
-	//TODO
+	good_data_t& good_data = get_good_data(good);
+	const std::lock_guard<std::mutex> lock_guard {*good_data.lock};
+	good_data.need_consumption_per_pop_type[&pop_type] += quantity;
```

Comment:
I see that you put the lock guard after getting the good data, at first I was worried this could cause problems but after thinking some more I realise this is indeed a clever way to minimise the amount of work that is limited by the lock, allowing multiple calls to do the (thread safe) good data lookup while waiting for their turn to modify its data.

```suggestion
	good_data_t& good_data = get_good_data(good);
	const std::lock_guard<std::mutex> lock_guard { *good_data.lock };
	good_data.need_consumption_per_pop_type[&pop_type] += quantity;
```
Also not urgent but it would be nice to have spaces in the `{}` constructor